utils/logger.go

```diff
@@ -0,0 +1,122 @@
+package utils
+
+import (
+	"strings"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// Logger is the wrapper of zap.SugaredLogger.
+// By default, the logger writes the output
+// to a JSON-format file.
+type Logger struct {
+	zLogger *zap.SugaredLogger
+}
+
+// A LoggerConfig contains the running environment
+// which is either "development" or "production",
+// and the path of file to write the logging output to.
+// Logging is enabled at DebugLevel and above in development mode,
+// at InfoLevel and above in production mode.
+type LoggerConfig struct {
+	Environment string `toml:"env"`
+	Path        string `toml:"path"`
+}
+
+func NewLogger(conf *LoggerConfig) *Logger {
+	zLevel := zap.NewAtomicLevel()
+	switch {
+	case strings.EqualFold("development", conf.Environment):
+		zLevel.SetLevel(zap.DebugLevel)
+	case strings.EqualFold("production", conf.Environment):
+		zLevel.SetLevel(zap.InfoLevel)
+	default:
+		panic("Environment must be either development or production")
+	}
+
+	zConfig := &zap.Config{
+		Level:             zLevel,
+		Development:       strings.EqualFold("development", conf.Environment),
+		Encoding:          "json",
+		DisableStacktrace: true, // turn off stacktrace capturing
+		EncoderConfig: zapcore.EncoderConfig{
+			TimeKey:        "timestamp",
+			LevelKey:       "level",
+			NameKey:        "logger",
+			CallerKey:      "path",
+			MessageKey:     "msg",
+			StacktraceKey:  "stack",
+			EncodeLevel:    zapcore.CapitalLevelEncoder,
+			EncodeTime:     zapcore.ISO8601TimeEncoder,
+			EncodeDuration: zapcore.StringDurationEncoder,
+		},
+		OutputPaths: []string{conf.Path},
+	}
+
+	logger, err := zConfig.Build()
+	if err != nil {
+		panic(err)
+	}
+	return &Logger{logger.Sugar()}
+}
+
+// Debug logs a message with some additional context
+// addressed by key-value pairs.
```

Comment:
The documentation for `Debug`, `Info`, `Warn` and `Error` is identical, which isn't super helpful. Is the difference between these logging levels really just the logging level itself? Maybe we could describe for which situations each logging level is intended? In other words, when does it make sense to call `Warn` vs `Info`? Alternatively, we could pass extra/less context information depending on the logging level? I guess I'm just trying to figure out a way that it's clear for developers (and also ourselves) how to choose which function to use. The same goes for `Panic` and `Fatal`.